File: Quick Sort/quick.js
    if (lista.length <= 1) {
        console.log("  ".repeat(nivel) + "Retorno:", lista);
        return lista;
    }
    let pivote = lista[0];
    let menores = lista.slice(1).filter(x => x <= pivote);
    let mayores = lista.slice(1).filter(x => x > pivote);
    console.log("  ".repeat(nivel) + `Pivote: ${pivote}, Menores: ${menores}, Mayores: ${mayores}`);
    return [...quickSort(menores, nivel + 1), pivote, ...quickSort(mayores, nivel + 1)];
}

let numeros = [8, 3, 1, 7, 0, 10, 2];
console.log("Original:", numeros);
let ordenado = quickSort(numeros);
console.log("Ordenado:", ordenado);

File: Quick Sort/quick.py
def quick_sort(lista, nivel=0):
    if len(lista) <= 1:
        print("  " * nivel + f"Retorno: {lista}")
        return lista
    pivote = lista[0]
    menores = [x for x in lista[1:] if x <= pivote]
    mayores = [x for x in lista[1:] if x > pivote]
    print("  " * nivel + f"Pivote: {pivote}, Menores: {menores}, Mayores: {mayores}")
    return quick_sort(menores, nivel + 1) + [pivote] + quick_sort(mayores, nivel + 1)

numeros = [8, 3, 1, 7, 0, 10, 2]
print("Original:", numeros)
ordenado = quick_sort(numeros)
print("Ordenado:", ordenado)